logic/standalone/aligner.py

```diff
@@ -0,0 +1,159 @@
+'''
+Aligner scans the whole level, then process all objects with name as "water_line".
+
+
+USAGE EXAMPLE:
+    aligner.AlignWater(playdo)
+'''
+
+import xml.etree.ElementTree as ET
+import logic.common.log_utils as log
+
+#--------------------------------------------------#
+'''Config'''
+
+# Enumerator
+STATE_INV = 0
+STATE_HOR = 1
+STATE_VER = 2
+
+# For logging
+CHAR_TRUE  = "  ○ "
+CHAR_FALSE = "  ☓ "
+
+
+
+#--------------------------------------------------#
+'''Public Functions'''
+
+def AlignWater(playdo):
+    log.Info("\n--- Aligning water_line Objects ---\n")
+
+    # Get objectgroup that contains the water
+    list_of_all_water_lines = playdo.GetAllObjectsWithName("water_line")
+    log.Must(f'Found {len(list_of_all_water_lines)} waterlines')
+
+    # Process individual object
+    count_need_alignment = 0
+    log.Info("--------------------------------------------------")
+    for water_line_obj in list_of_all_water_lines:
+        if water_line_obj != list_of_all_water_lines[0]:
+            log.Extra("--------------------------------------------------")
+
+        # Obtain vertice data
+        polyline_attribute = water_line_obj.find('polyline')
+        if polyline_attribute == None : continue
+        points_string = polyline_attribute.get('points')
+        line_tuple = GetVertexTuple(points_string)
+
+        # Check
+        obj_state = CheckNeedAlignment(line_tuple)
+        if obj_state == STATE_INV: continue
+        count_need_alignment += 1
+        new_vertice_str = MakeAlignedStr(obj_state, line_tuple)
+
+        # Apply changes
+        polyline_attribute.set('points', new_vertice_str)
```

Comment:
The big change I'd like is that you refactor this so that we don't need obj_state at all. The passing around of this enum feels a little clumsy and also results in some redundant work. Notice how you unpack the values of line_tuple twice?

How about a function called MakeNewAlignment() which does the work of both CheckNeedAlignment and MakeAlignedStr. If there is an alignment to make, it returns the new aligned string. And if there is no alignment to make, it returns None.

We can then inspect that value here (for None or if it's a new string) and increment counters /apply changes as needed